exercicios/para-casa/entregas/andrea-vetter/BankAccount.js

```diff
@@ -0,0 +1,160 @@
+import { Bank, bank1, bank2 } from "./Bank.js"
+import { Client, client1 } from "./Client.js"
+
+class BankAccount {
+  client;
+  bank;
+  accountNumber;
+  agencyNumber;
+  #balance = 0;
+
+  constructor(client, bank, accountNumber, agencyNumber) {
+    if (!(client instanceof Client)) {
+      throw "A pessoa cliente deve estar registrada no cadastro de clientes!";
+    }
+
+    if (!(bank instanceof Bank)) {
+      throw "O banco deve estar registrado no cadastro de bancos!";
+    }
+
+    this.client = client;
+    this.bank = bank;
+    this.accountNumber = accountNumber;
+    this.agencyNumber = agencyNumber;
+  }
+
+  get balance() {
+    return this.#balance;
+  }
+
+  set balance(newBalance) {
+    this.#balance = newBalance;
+  }
+
+  creditAmount(amount) {
+    this.#balance += amount;
+  }
+
+  debitAmount(amount) {
+    if (this.#balance < amount) {
+      throw new Error("Não há crédito suficiente em sua conta!");
+    }
+    this.#balance -= amount;
+  }
+
+  transferTo(anotherAccount, amount) {
+    if (!(anotherAccount instanceof BankAccount)) {
+      throw new Error("Essa conta não existe!");
+    }
+    if (this.bank === anotherAccount.bank) {
```

Comment:
Por curiosidade, isso está funcionando? Pois como bank é uma instancia de Bank você não especifica qual dos parâmetros devem ser comparados. Como sugestão, você poderia tentar dessa forma: 

`if (this.bank.bankCode === anotherAccount.bank.bankCode)`